### ISSUE TITLE
KeyError due to API change

### ISSUE THREAD

**Issue by foo-git** [foo-git/rewe-discounts#8]
At first it looks like the API output has changed and the program cannot deal correctly with the output. After executing the program, it fails with this error message:
```
 Traceback (most recent call last):
  File "/x/rewe_discounts", line 193, in <module>
    categories = data['_meta']['categories']
 KeyError: '_meta'
```

However the API url (https://mobile-api.rewe.de/products/offer-search) yields a 500 error, which looks like an internal error of the rewe api. Not sure if I can fix this:
```
errorReason: "status - 500 by calling Endpoint Service, because of reason: result.data.map is not a function}"
```

**Comment by torbentee** [comment]
The new url for offers is `https://mobile-api.rewe.de/api/v3/all-offers?marketCode=<market-id>`. But the data structure also changed.

**Comment by foo-git** [comment]
Thanks @torbentee for providing the url. How did you discover it?

I checked the data structure and as you mentioned, it changed. While the new categorization scheme looks somewhat straightforward to apply, some product details like the base price of an example product below (e.g. 1 kg = 2,32) is now inside the subtitle string, so I need to apply some regex-fu. Additionally, there are some interesting product details available like priceAverage, where I don't know yet the significance of the number.
![screenshot](https://user-images.githubusercontent.com/53096886/171345356-5664fb4d-a62a-4352-ad4e-ec5243de9b35.png)

**Comment by torbentee** [comment]
@foo

> Thanks @torbentee for providing the url. How did you discover it?

You're welcome. I intercepted requests from the Rewe iOS app with [proxyman](https://proxyman.io/)

**Comment by foo-git** [comment]
This should be fixed in [2f7d281](https://github.com/foo-git/rewe-discounts/commit/2f7d2810f68cfa23277cc9d247cc0460cc90f7cc).
